src/main.cpp

```diff
@@ -3109,6 +3115,10 @@ bool static LoadBlockIndexDB()
             pindex->BuildSkip();
         if (pindex->IsValid(BLOCK_VALID_TREE) && (pindexBestHeader == NULL || CBlockIndexWorkComparator()(pindexBestHeader, pindex)))
             pindexBestHeader = pindex;
+
+        // Insert into versionbits block rule index and recompute soft fork deployment states for chain
+        Consensus::VersionBits::BlockRuleIndex& blockRuleIndex = g_blockRuleIndex;
```

Comment:
Why not on top of the function (like CChainParams& chainparams, like it was in my code) ready to become a parameter for the function?
Why did you spend any time in moving it down?